string testname
---
bool success
string message